app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ExposureNotificationRepository.kt

```diff
@@ -222,4 +242,17 @@ class ExposureNotificationRepositoryImpl(
                 }
         }
     }
+
+    override fun getTemporaryExposureKeys() {
+        temporaryExposureKeySubject.loading()
+        exposureNotificationClient.temporaryExposureKeyHistory
+            .addOnSuccessListener {
+                temporaryExposureKeySubject.loaded(it)
+                temporaryExposureKeySubject.idle()
```

Comment:
Maybe idle is not exactly what you want here, is it? Once you clear the loaded state by idle and in case nobody is observing, the data will be lost.